src/Parsers/ASTAlterQuery.cpp

```diff
@@ -127,364 +127,368 @@ const char * ASTAlterCommand::typeToString(ASTAlterCommand::Type type)
     UNREACHABLE();
 }
 
-void ASTAlterCommand::formatImpl(const FormatSettings & settings, FormatState & state, FormatStateStacked frame) const
+void ASTAlterCommand::formatImpl(FormattingBuffer out) const
 {
     if (type == ASTAlterCommand::ADD_COLUMN)
     {
-        settings.ostr << (settings.hilite ? hilite_keyword : "") << "ADD COLUMN " << (if_not_exists ? "IF NOT EXISTS " : "")
-                      << (settings.hilite ? hilite_none : "");
-        col_decl->formatImpl(settings, state, frame);
+        out.writeKeyword("ADD COLUMN ");
+        out.writeKeyword(if_not_exists ? "IF NOT EXISTS " : "");
```

Comment:
We don't have to call `writeKeyword()` if `if_not_exists == false`:

```
if (if_not_exists)
    out.writeKeyword("IF NOT EXISTS ")
```

Reply:
This pattern of `out.writeKeyword(condition ? "a" : "b")` is used throughout the code, and I believe it's a nice and consistent way to output numerous optional modifiers. It's shorter in one line, and the odd `if` won't be breaking the flow of multiple writeKeywords visually.

Reply:
I don't see any sense in writing `hilite_keyword` and `hilite_none` with nothing in between.

Reply:
You could at least modify `writeKeyword()` and make it skip its argument if it's empty.